CeeFM/Decideify
# Subscribe buttons now live and creating/deleting from database

**Description**
You can now subscribe to users, at least in terms of adding the appropriate data to the databases subscriptions table, and rendering the correct buttons (subscribe when a subscription doesn't exist for that user yet, and unsubscribe when one does, with create and delete functionality, respectively)

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- On social.js you can:
  - Click on any post to open the post details into a modal, where you can view comments and non-truncated titles and post contents (main page ones are truncated to ensure styling consistency)
  - Clicking a Subscribe button should add the currently logged in user's id as the SubscribeUserProfileId and the post authors UserProfileId and store it in the Subscriptions table of the database
  - Clicking Unsubscribe button will remove the existing entry in the database matching the above conditions

Checklist
